spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpoint.java

```diff
@@ -101,14 +124,38 @@ protected Object handleRequestMessage(Message<?> requestMessage) {
 		String exchangeName = generateExchangeName(requestMessage);
 		String routingKey = generateRoutingKey(requestMessage);
 		if (this.expectReply) {
-			return this.sendAndReceive(exchangeName, routingKey, requestMessage, correlationData);
+			return sendAndReceive(exchangeName, routingKey, requestMessage, correlationData);
 		}
 		else {
-			this.send(exchangeName, routingKey, requestMessage, correlationData);
+			send(exchangeName, routingKey, requestMessage, correlationData);
+			if (this.waitForConfirm && correlationData != null) {
+				waitForConfirm(requestMessage, correlationData);
+			}
 			return null;
 		}
 	}
 
+	private void waitForConfirm(Message<?> requestMessage, CorrelationData correlationData) {
+		try {
+			Confirm confirm = correlationData.getFuture().get(getConfirmTimeout().toMillis(), TimeUnit.MILLISECONDS);
+			if (!confirm.isAck()) {
+				throw new MessageHandlingException(requestMessage, "Negative publisher confirm received: " + confirm);
```

Comment:
I have fought recently about a `BeanDefinition` context in our exceptions. I think it is not going to happen in the super class automatically because you throw already `MessageHandlingException`. 
Please, consider to include `this` for presenting a `BeanDefinition` context in the exception message for better end-user experience. Or just throw non-MessageHandlingExceptions!